homeassistant/components/geocaching/entity.py

```diff
@@ -0,0 +1,54 @@
+"""Base entities for Geocaching devices."""
+
+from typing import cast
+
+from geocachingapi.models import GeocachingCache
+
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import CACHE_ID_SENSOR_FORMAT, DOMAIN
+from .coordinator import GeocachingDataUpdateCoordinator
+
+
+# A device can have multiple entities, and for a cache which requires multiple entities we want to group them together.
+# Therefore, we create a device for each cache, which holds all related entities.
+# This function returns the device info for a cache.
+def get_cache_device_info(cache: GeocachingCache) -> DeviceInfo:
+    """Generate device info for a cache."""
+    return DeviceInfo(
+        name=f"Geocache {cache.reference_code}",
+        identifiers={(DOMAIN, cast(str, cache.reference_code))},
+        entry_type=DeviceEntryType.SERVICE,
+        manufacturer="Groundspeak, Inc.",
+    )
```

Comment:
Why is this here? And not just part of the base entity?

A method that just returns a single thing, can just be inlined.